### PR TITLE
[PLAYER-239] Handle ad request errors and report correct playhead offset

### PR DESCRIPTION
**Pull request by dchauOoyala**

### CODE REVIEW

File: js/ssai_pulse.js [hunk 1]
```diff
@@ -472,6 +495,11 @@ OO.Ads.manager(function(_, $)
     this.onRequestError = function()
     {
       OO.log("SSAI Pulse: Error");
+      if (_.has(this.adIdDictionary, this.currentId3Object.adId))
```
Comment (aeng7) [inline js/ssai_pulse.js]: Can we add a null check for this.currentId3Object?

File: js/ssai_pulse.js [hunk 2]
```diff
@@ -593,7 +621,7 @@ OO.Ads.manager(function(_, $)
      */
     var _appendAdsProxyQueryParameters = _.bind(function(url, adId)
     {
-      var offset = OFFSET_PARAM + OFFSET_VALUE;
```
Comment (aeng7) [inline js/ssai_pulse.js]: Do we need OFFSET_VALUE anymore? If it is no longer used, we should remove its declaration.

**Comment by pilievOoyala** [comment]
LGTM.

File: js/ssai_pulse.js [hunk 3]
```diff
@@ -43,6 +43,7 @@ OO.Ads.manager(function(_, $)
     this.currentAd = null;
 
     var amc  = null;
+    var currentOffset = 0;
```
Comment (aeng7) [inline js/ssai_pulse.js]: Do we need to reinitialize currentOffset anywhere (like on replays, content change, etc)?

File: test/unit-tests/ssai_pulse_test.js [hunk 4]
```diff
@@ -614,4 +614,99 @@ describe('ad_manager_ssai_pulse', function()
...
+    offset = SsaiPulse.getCurrentOffset();
+    expect(offset).to.be(0);
+
+    SsaiPulse.onPlayheadTimeChanged(eventName, playhead, null, {});
+    offset = SsaiPulse.getCurrentOffset();
+    expect(offset).to.be(0);
+
+    SsaiPulse.onPlayheadTimeChanged(eventName, playhead, {}, {});
+    offset = SsaiPulse.getCurrentOffset();
+    expect(offset).to.be(0);
```
Comment (aeng7) [inline test/unit-tests/ssai_pulse_test.js]: Nice negative tests!

File: js/ssai_pulse.js [hunk 5]
```diff
@@ -555,7 +598,7 @@ OO.Ads.manager(function(_, $)
 
     var _onContentChanged = function()
     {
-      // Callback for example listener registered in this.initialize
+      currentOffset = 0;
```
Comment (aeng7) [inline js/ssai_pulse.js]: Should we set currentAd and currentId3Object to null as well here?

Comment (dchauOoyala) [inline js/ssai_pulse.js]: done

File: js/ssai_pulse.js (hunk 2 above)
Comment (dchauOoyala) [inline js/ssai_pulse.js]: done

File: js/ssai_pulse.js (hunk 3 above)
Comment (dchauOoyala) [inline js/ssai_pulse.js]: added resetting of values onContentChanged and onReplay

**Comment by aeng7** [comment]
LGTM!

**Pull request merged**